yt_dlp/extractor/youtube.py

```diff
@@ -752,32 +948,54 @@ def extract_ytcfg(self, video_id, webpage):
                 default='{}'), video_id, fatal=False) or {}
 
+    def _generate_oauth_headers(self):
+        self._update_oauth()
+        oauth_token = YoutubeBaseInfoExtractor._OAUTH_ACCESS_TOKEN_CACHE.get(self._OAUTH_PROFILE)
+        if not oauth_token:
+            return {}
+
+        return {
+            'Authorization': f'{oauth_token["token_type"]} {oauth_token["access_token"]}',
+        }
+
+    def _generate_cookie_auth_headers(self, *, ytcfg=None, account_syncid=None, session_index=None, origin=None, **kwargs):
+        headers = {}
+        account_syncid = account_syncid or self._extract_account_syncid(ytcfg)
+        if account_syncid:
+            headers['X-Goog-PageId'] = account_syncid
+        if session_index is None:
+            session_index = self._extract_session_index(ytcfg)
+        if account_syncid or session_index is not None:
+            headers['X-Goog-AuthUser'] = session_index if session_index is not None else 0
+
+        auth = self._generate_sapisidhash_header(origin)
+        if auth is not None:
+            headers['Authorization'] = auth
+            headers['X-Origin'] = origin
+
+        return headers
+
     def generate_api_headers(
             self, *, ytcfg=None, account_syncid=None, session_index=None,
-            visitor_data=None, identity_token=None, api_hostname=None, default_client='web'):
+            visitor_data=None, api_hostname=None, default_client='web', **kwargs):
 
         origin = 'https://' + (self._select_api_hostname(api_hostname, default_client))
         headers = {
             'X-YouTube-Client-Name': str(
                 self._ytcfg_get_safe(ytcfg, lambda x: x['INNERTUBE_CONTEXT_CLIENT_NAME'], default_client=default_client)),
             'X-YouTube-Client-Version': self._extract_client_version(ytcfg, default_client),
             'Origin': origin,
-            'X-Youtube-Identity-Token': identity_token or self._extract_identity_token(ytcfg),
-            'X-Goog-PageId': account_syncid or self._extract_account_syncid(ytcfg),
             'X-Goog-Visitor-Id': visitor_data or self._extract_visitor_data(ytcfg),
             'User-Agent': self._ytcfg_get_safe(ytcfg, lambda x: x['INNERTUBE_CONTEXT']['client']['userAgent'], default_client=default_client),
+            **self._generate_oauth_headers(),
+            **self._generate_cookie_auth_headers(ytcfg=ytcfg, account_syncid=account_syncid, session_index=session_index, origin=origin),
         }
-        if session_index is None:
-            session_index = self._extract_session_index(ytcfg)
-        if account_syncid or session_index is not None:
-            headers['X-Goog-AuthUser'] = session_index if session_index is not None else 0
-
-        auth = self._generate_sapisidhash_header(origin)
-        if auth is not None:
-            headers['Authorization'] = auth
-            headers['X-Origin'] = origin
         return filter_dict(headers)
 
+    @property
+    def _webpage_headers(self):
```

Comment:
on 2nd thought, better to stick with the existing convention
```suggestion
    def _generate_webpage_headers(self):
```